ibm/resource_ibm_is_instance_network_interface.go

```diff
@@ -0,0 +1,649 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return resourceIBMIsInstanceNetworkInterfaceRead(context, d, meta)
+}
+
+func resourceIBMIsInstanceNetworkInterfaceDelete(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(ClientSession).VpcV1API()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deleteInstanceNetworkInterfaceOptions := &vpcv1.DeleteInstanceNetworkInterfaceOptions{}
+
+	parts, err := sepIdParts(d.Id(), "/")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instance_id := parts[0]
+	network_intf_id := parts[1]
+	isNICKey := "instance_network_interface_key_" + instance_id
+	ibmMutexKV.Lock(isNICKey)
+	defer ibmMutexKV.Unlock(isNICKey)
+
+	deleteInstanceNetworkInterfaceOptions.SetInstanceID(instance_id)
+	deleteInstanceNetworkInterfaceOptions.SetID(network_intf_id)
+
+	response, err := vpcClient.DeleteInstanceNetworkInterfaceWithContext(context, deleteInstanceNetworkInterfaceOptions)
+	if err != nil {
+		log.Printf("[DEBUG] DeleteInstanceNetworkInterfaceWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("DeleteInstanceNetworkInterfaceWithContext failed %s\n%s", err, response))
+	}
+
+	_, err = isWaitForNetworkInterfaceDelete(vpcClient, d.Id(), d.Timeout(schema.TimeoutUpdate), d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("Error occured while waiting for network interface %s", err))
+	}
+
+	_, err = isWaitForInstanceAvailable(vpcClient, instance_id, d.Timeout(schema.TimeoutCreate), d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func isWaitForNetworkInterfaceAvailable(vpcClient *vpcv1.VpcV1, id string, timeout time.Duration, d *schema.ResourceData) (interface{}, error) {
+	log.Printf("Waiting for dedicated host (%s) to be available.", id)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{isNetworkInterfacePending},
+		Target:     []string{isNetworkInterfaceAvailable, isNetworkInterfaceFailed},
+		Refresh:    isNetworkInterfaceRefreshFunc(vpcClient, id, d),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+
+	return stateConf.WaitForState()
+}
+
+func isNetworkInterfaceRefreshFunc(vpcClient *vpcv1.VpcV1, id string, d *schema.ResourceData) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		getInstanceNetworkInterfaceOptions := &vpcv1.GetInstanceNetworkInterfaceOptions{}
+		parts, err := sepIdParts(d.Id(), "/")
+		if err != nil {
+			return nil, "", fmt.Errorf("Error splitting ID in parts %s", err)
+		}
+
+		getInstanceNetworkInterfaceOptions.SetInstanceID(parts[0])
+		getInstanceNetworkInterfaceOptions.SetID(parts[1])
+
+		networkInterface, response, err := vpcClient.GetInstanceNetworkInterface(getInstanceNetworkInterfaceOptions)
+		if err != nil {
+			return nil, "", fmt.Errorf("GetInstanceNetworkInterface failed %s\n%s", err, response)
+		}
+		d.Set("status", *networkInterface.Status)
```

Comment:
why is status set in wait logic?

Reply:
we set the status in Instance resource refresh method as well. To keep the status up to current. is it not needed ?